File: docs/breez-sdk/snippets/kotlin_mpp_lib/shared/src/commonMain/kotlin/com/example/kotlinmpplib/BuyingBitcoin.kt
package com.example.kotlinmpplib

import breez_sdk_spark.*

class BuyingBitcoin {
    suspend fun buyBitcoin(sdk: BreezSdk) {
        // ANCHOR: buy-bitcoin
        // Optionally, lock the purchase to a specific amount
        val optionalLockedAmountSat: ULong = 100_000u
        // Optionally, set a redirect URL for after the purchase is completed
        val optionalRedirectUrl = "https://example.com/purchase-complete"

        val request = BuyBitcoinRequest(
            provider = BuyBitcoinProvider.MOONPAY,
            lockedAmountSat = optionalLockedAmountSat,
            redirectUrl = optionalRedirectUrl
        )

        val response = sdk.buyBitcoin(request)
        // Log.v("Breez", "Open this URL in a browser to complete the purchase:")
        // Log.v("Breez", "${response.url}")
        // ANCHOR_END: buy-bitcoin
    }

    suspend fun buyBitcoinViaCashapp(sdk: BreezSdk) {
        // ANCHOR: buy-bitcoin-cashapp
        val request = BuyBitcoinRequest(
            provider = BuyBitcoinProvider.CASH_APP
        )

        val response = sdk.buyBitcoin(request)
        // Log.v("Breez", "Open this URL in Cash App to complete the purchase:")
        // Log.v("Breez", "${response.url}")
        // ANCHOR_END: buy-bitcoin-cashapp
    }
}

File: docs/breez-sdk/snippets/swift/BreezSdkSnippets/Sources/BuyingBitcoin.swift
import BreezSdkSpark

func buyBitcoin(sdk: BreezSdk) async throws {
    // ANCHOR: buy-bitcoin
    // Optionally, lock the purchase to a specific amount
    let optionalLockedAmountSat: UInt64 = 100_000
    // Optionally, set a redirect URL for after the purchase is completed
    let optionalRedirectUrl = "https://example.com/purchase-complete"

    let request = BuyBitcoinRequest.moonpay(
        lockedAmountSat: optionalLockedAmountSat,
        redirectUrl: optionalRedirectUrl
    )

    let response = try await sdk.buyBitcoin(request: request)
    print("Open this URL in a browser to complete the purchase:")
    print("\(response.url)")
    // ANCHOR_END: buy-bitcoin
}

func buyBitcoinViaCashapp(sdk: BreezSdk) async throws {
    // ANCHOR: buy-bitcoin-cashapp
    let request = BuyBitcoinRequest.cashApp(amountSats: nil)

    let response = try await sdk.buyBitcoin(request: request)
    print("Open this URL in Cash App to complete the purchase:")
    print("\(response.url)")
    // ANCHOR_END: buy-bitcoin-cashapp
}
